pulp_2to3_migration/app/pre_migration.py

```diff
@@ -447,34 +446,38 @@
                                         fields=['not_in_pulp2'],
                                         batch_size=1000)
 
-    # Mark importers
-    mongo_imp_object_ids = set(str(i.id) for i in Importer.objects.only('id'))
-    premigrated_imps = Pulp2Importer.objects.filter(pulp2_repository__type__in=psql_repo_types)
-    premigrated_imp_object_ids = set(premigrated_imps.values_list('pulp2_object_id',
-                                                                  flat=True))
-    removed_imp_object_ids = premigrated_imp_object_ids - mongo_imp_object_ids
-
-    removed_imps = []
-    for pulp2importer in Pulp2Importer.objects.filter(pulp2_object_id__in=removed_imp_object_ids):
-        pulp2importer.not_in_pulp2 = True
-        removed_imps.append(pulp2importer)
-
-    Pulp2Importer.objects.bulk_update(objs=removed_imps,
-                                      fields=['not_in_pulp2'],
-                                      batch_size=1000)
-
-    # Mark distributors
-    mongo_dist_object_ids = set(str(i.id) for i in Distributor.objects.only('id'))
-    premigrated_dists = Pulp2Distributor.objects.filter(pulp2_repository__type__in=psql_repo_types)
-    premigrated_dist_object_ids = set(premigrated_dists.values_list('pulp2_object_id',
-                                                                    flat=True))
-    removed_dist_object_ids = premigrated_dist_object_ids - mongo_dist_object_ids
 
-    removed_dists = []
-    for pulp2dist in Pulp2Distributor.objects.filter(pulp2_object_id__in=removed_dist_object_ids):
-        pulp2dist.not_in_pulp2 = True
-        removed_dists.append(pulp2dist)
+async def delete_old_resources(plan):
+    """
+    Delete pre-migrated and migrated data for resources which are migrated fully on every run.
+
+    Delete pre-migrated importers and distributors, migrated Remotes, Publications,
+    and Distributions according to a specified Migration Plan.
+
+    Args:
+        plan(MigrationPlan): A Migration Plan
 
-    Pulp2Distributor.objects.bulk_update(objs=removed_dists,
-                                         fields=['not_in_pulp2'],
-                                         batch_size=1000)
+    """
+
+    for plugin in plan.get_plugin_plans():
+        with transaction.atomic():
+            for importer_type, importer_migrator in plugin.migrator.importer_migrators.items():
+                # delete all pre-migrated importers for this plugin
+                Pulp2Importer.objects.filter(pulp2_type_id=importer_type).delete()
+                # delete all migrated Remotes for this plugin
+                for remote_model in importer_migrator.pulp3_remote_models:
+                    remote_model.objects.all().delete()
+
+                # TODO: since all Remotes are removed, RemoteArtifacts need to be updated after
+                #  new Remotes are created. Maybe it's worth to keep track of Remotes and not
+                #  wipe them out every time?
```

Comment:
Still a question, how to update/add RemoteArtifacts without going through all the content all the time.